bottlerocket-settings-models/modeled-types/src/shared.rs

```diff
@@ -945,6 +945,90 @@ string_impls_for!(Lockdown, "Lockdown");
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// ApiCommand represents a valid Bottlerocket API Command. It stores the command as a vector of
+/// strings and ensures that the first argument is apiclient.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Default, Serialize, Deserialize)]
+pub struct ApiCommand(pub Vec<String>);
```

Comment:
Why does the inner type need to be `pub`?
```suggestion
pub struct ApiCommand(Vec<String>);
```

Reply:
Had kept it public to be able to access the values. In the current revision, I have added get_args function to fetch the arguments.